fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -253,51 +252,56 @@ private boolean shouldClearExistingIndexEntries(@Nonnull IndexState indexState,
     }
 
     @Nonnull
-    public IndexingBase setFallbackMode() {
-        fallbackMode = true;
-        return this;
+    public void setFallbackMode() {
+        forceStampOverwrite = true; // must overwrite a previous indexing method's stamp
     }
 
     @Nonnull
     private CompletableFuture<Void> setIndexingTypeOrThrow() {
-        byte[] stamp = getTypeStamp();
-        final List<Object> additionalLogMessageKeyValues = Arrays.asList(LogMessageKeys.CALLING_METHOD, "setIndexingTypeOrThrow",
-                LogMessageKeys.LOCAL_VERSION, stamp);
-
-        return buildCommitRetryAsync( (store, ignore) -> {
+        return getRunner().runAsync(context -> openRecordStore(context).thenCompose(store -> {
             Transaction transaction = store.getContext().ensureActive();
+            byte[] stamp = getIndexingTypeStamp();
             byte[] stampKey = indexBuildTypeSubspace(store, common.getIndex()).getKey();
```

Comment:
I think getIndexingTypeStamp can also return IndexMetaDataProto.IndexMetaDataIndexingStamp, and in setIndexingTypeOrThrow you can compare two protobufs (which can provide more detailed differences) rather than two byte arrays by converting the stored stamp to protobuf.